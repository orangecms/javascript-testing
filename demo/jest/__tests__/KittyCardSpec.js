import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-addons-test-utils';
import KittyCard from '../components/KittyCard/KittyCard';

jest.unmock('../components/KittyCard/KittyCard');

// dummy
const kitty = { _id: 123, name: 'Miss Kitty Fantastico' };

describe('KittyCard', () => {
  it('shows the name of the cat', () => {
    const kittyCard = TestUtils.renderIntoDocument(
      <KittyCard kitty={kitty} />
    );
    const renderedKittyCard = ReactDOM.findDOMNode(kittyCard);
    expect(renderedKittyCard.textContent).toMatch(/Miss Kitty Fantastico/);
  });

  it('shows the number of the cat', () => {
    const kittyCard = TestUtils.renderIntoDocument(
      <KittyCard kitty={kitty} />
    );
    const renderedKittyCard = ReactDOM.findDOMNode(kittyCard);
    expect(renderedKittyCard.textContent).toMatch(/is cat #123/);
  });
});
