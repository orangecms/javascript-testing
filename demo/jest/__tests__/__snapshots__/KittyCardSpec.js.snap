// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KittyCard renders correctly 1`] = `
<figure
  style={
    Object {
      "border": "5px solid purple",
      "font": "bold 24pt Roboto-Regular, sans-serif",
      "margin": 5,
      "padding": 10,
    }
  }
>
  <img
    src="http://thecatapi.com/api/image/get?meow=123"
  />
  <figcaption>
    Miss Kitty Fantastico
     is cat #
    123
     
    ðŸ˜¸
  </figcaption>
</figure>
`;
